[17.0][MIG] auto_backup: Migration to 17.0

rpinset opened the pull request.

hbrunn committed:
[ADD] make retrieval configurable per folder; use msgid to exclude
duplicates; don't break existing configurations

hbrunn committed:
[MRG] lp:~therp-nl/therp-addons/fetchmail_attach_from_folder rev 89

NL66278 committed:
[IMP] - convert mailadresses to lowercase to increase change for match
      - handle situation that optional mail address componenents not present in mail
      - slight reorganisation of code to aid debugging and made pep8 compliant as well.

hbrunn committed:
[IMP] autopep8
[IMP] more efficient sorting of algorithms
[IMP] improve readability of help strings

hbrunn committed:
[ADD] msg_state field for folders - set the state of messages fetched
from imap servers
[IMP] clean up the folders' treeview

hbrunn committed:
[IMP] refactored fetch_mail
[IMP] also log when we're done with one folder

hbrunn committed:
[FIX] attach mail attachments to matched object in non-OpenERP case

hbrunn committed:
[IMP] make fetch_mail's refactoring more useful by returning the ids of
mails creates/objects matched
[ADD] preliminary docstrings

hbrunn committed:
[FIX] don't rollback() transation, use savepoint. This fixes concurrency
issues with long running cron threads as rollback also releases the lock

gurneyalex committed:
[REVERT] merged and pushed on 7.0 instead of 6.1 branch...

hbrunn committed:
[ADD] port fetchmail_attach_from_folder to 7.0

hbrunn committed:
[IMP] remove deprecated type field for views

hbrunn committed:
[IMP] adjust view definitions to 7.0
[FIX] use mail_thread.message_parse in wizard

sbidoul committed:
move all modules to __unported__ on master branch and set installable=False

hbrunn committed:
[REN] preliminary migration of fetchmail_attach_from_folder

hbrunn committed:
[IMP] make manual attach wizard usable again

hbrunn committed:
[ADD] placeholders

hbrunn committed:
[FIX] rename OpenERP->Odoo

hbrunn committed:
[IMP] readme

hbrunn committed:
[IMP] use a dedicated logger

hbrunn committed:
[FIX] we want object_id not to be required

hbrunn committed:
[ADD] test matching algorithms

yvaucher committed:
Add license key in __openerp__.py

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

Commit:
Remove auto_install on fetchmail_attach_from_folder

The key 'active' is deprecated and is a synonym for 'auto_install'.
This addon should not be automatically installed.

yvaucher committed:
Add bug tracker link on README.rst

hbrunn committed:
[FIX] honor sequence

hbrunn committed:
[ADD] allow to deactivate folders

Commit:
Add module datetime_formatter.

This module allows the developer to format any date/time combination in
the most comfortable way for the user, just with a quick call to

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

StefanRijnhart committed:
[FIX] Some fields are not there if the client asks for a search form

StefanRijnhart committed:
[RFR] Override hook

StefanRijnhart committed:
[IMP] Full email preview when attaching manually

gurneyalex committed:
[SEC] fetchmail_attach_from_folder: fix unsafe eval

pedrobaeza committed:
[MIG] Rename manifest files

cubells committed:
datatime_formatter migration to 9.0

OCA Transbot updated translations from Transifex

yajo committed:
[MIG][10.0][datetime_formatter] Migration

Special highlights:

- Fixed some license headers.
- Adhered to normal addon structure.
- Removed `exceptions.py` and use `UserError` instead.
- Allow submodules to import some constants from main namespace.

The rest is basically a standard migration.

OCA Transbot updated translations from Transifex

Commit:
[MIG] datetime_formatter: Migration to 11.0

[UPD] Update datetime_formatter.pot

NL66278 committed:
[MIG] fetchmail_attach_from_folder. Migration to 10.0.

NL66278 committed:
[FIX] fetchmail_attach_from_folder. Refactor for more testing.

NL66278 committed:
[FIX] fetchnail_attach_from_folder. Can not pass object to action context.

NL66278 committed:
[IMP] fetchmail_attach_from_folder. Separate confirm for folder.

Now it is impossible to confirm a connection to a imap server if there is a problem
in one folder. This commit provides a new state field on the folder and e separate
comfirm button per folder.

NL66278 committed:
[IMP] fetchmail_attach_from_folder. Show available folders.

Also do not reset connection status on change of underlying folders.

NL66278 committed:
[IMP] fetchmail_attach_from_folder. Button to retrieve single folder.

Also possibility to reset state for folder.

etobella committed:
[FIX] module name

[UPD] Update base_remote.pot

etobella committed:
[FIX] missing super declaration

etobella committed:
Added translation using Weblate (Español (España))

etobella committed:
[FIX] base_remote: Views

etobella committed:
[IMP] name of remotes should be case insensitive

etobella committed:
[FIX] base_remote: Do not raise Error on login failure

[ADD] icon.png

ernestotejeda committed:
[MIG] datetime_formatter: Migration to 12.0

[UPD] Update datetime_formatter.pot

[UPD] README.rst

Added translation using Weblate (Portuguese)

Translated using Weblate (Portuguese)

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-12.0/server-tools-12.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-datetime_formatter/pt/

Added translation using Weblate (Chinese (Simplified))

Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-12.0/server-tools-12.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-datetime_formatter/zh_CN/

Translated using Weblate (Croatian)

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-12.0/server-tools-12.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-datetime_formatter/hr/

Commit:
[12.0][MIG] base_remote

[UPD] Update base_remote.pot

oca-travis committed:
[UPD] Update fetchmail_attach_from_folder.pot

ntsirintanis committed:
[FIX] replace simplejson dependency with json

OCA-git-bot committed:
fetchmail_attach_from_folder 10.0.1.1.0

astirpe committed:
[13.0][ADD] base_changeset

Commit:
[12.0][FIX] base_remote: search =ilike

base_remote 12.0.1.0.1

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-base_remote/

Commit:
[IMP] base_remote: black, isort, prettier

Commit:
[13.0][MIG] base_remote

[UPD] Update base_remote.pot

victoralmau committed:
[IMP] datetime_formatter: black, isort, prettier

victoralmau committed:
[MIG] datetime_formatter: Migration to 13.0

[UPD] Update datetime_formatter.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-datetime_formatter/

anasuarez1 committed:
Added translation using Weblate (Spanish)

legalsylvain committed:
[IMP] database_cleanup : delete also obsolete SQL views

Co-authored-by: Pedro M. Baeza <pedro.baeza@tecnativa.com>

fshah-initos committed:
[IMP] isort, black, prettier.

fshah-initos committed:
[MIG] Migrate module base_remote to v14.

[UPD] Update base_remote.pot

[UPD] README.rst

[UPD] Update base_remote.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-base_remote/

dsolanki-initos committed:
[MIG] Migrate module datetime_formatter to v14.

[UPD] Update datetime_formatter.pot

[UPD] README.rst

dsolanki-initos committed:
[IMP] base_changeset: isort, black, prettier.

dsolanki-initos committed:
[MIG] Migrate module base_changeset to v14.

hbrunn committed:
[IMP] base_changeset: Make it impossible to set context key from outside

hbrunn committed:
[IMP] base_changeset: Inhibit creating records with restricted fields too

hbrunn committed:
[ADD] base_changeset: Determine if a rule applies by an expression

hbrunn committed:
[ADD] base_changeset: hbrunn as contributor

hbrunn committed:
[IMP] base_changeset: overhaul security

Commit:
[FIX] base_changeset: fix write return type

StefanRijnhart committed:
[IMP] base_changeset: black, isort, prettier

StefanRijnhart committed:
[FIX] base_changeset: assign values for all computed fields

legalsylvain committed:
[FIX] base_changeset : fix images path in readme files

StefanRijnhart committed:
[FIX] base_changeset: models may not have a `name` field

Fixes

```
bank.changeset_ids.name_get()
*** AttributeError: 'res.partner.bank' object has no attribute 'name'
```

primes2h committed:
Added translation using Weblate (Italian)

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 41.0% (39 of 95 strings)

Translation: server-tools-14.0/server-tools-14.0-base_changeset
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-base_changeset/it/

Commit:
[IMP] base_changeset: optionally prevent self validation

TautvydasBanevicius committed:
[IMP] datetime_formatter: black, isort, prettier, pylint, flake8.

Modified exception on method best_match according flake8 warning.
Modified test method test_user with_context(**) according pyling warning.

[UPD] Update datetime_formatter.pot

[UPD] README.rst

Added translation using Weblate (Spanish (Argentina))

Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-15.0/server-tools-15.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-datetime_formatter/es_AR/

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-15.0/server-tools-15.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-datetime_formatter/

TautvydasBanevicius committed:
[MIG] datetime_formatter: Bumped version to 15.0

ofonseca-pyming committed:
Added translation using Weblate (Catalan)

ofonseca-pyming committed:
Translated using Weblate (Catalan)

Currently translated at 1.0% (1 of 96 strings)

Translation: server-tools-14.0/server-tools-14.0-base_changeset
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-base_changeset/ca/

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (22 of 22 strings)

Translation: server-tools-14.0/server-tools-14.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-base_remote/es_AR/

olgamarcocb committed:
[MIG] base_remote: Migration to 15.0

Co-authored-by: Enric Tobella <etobella@creublanca.es>

StefanRijnhart committed:
[RFR] base_changeset: adapt intercept of create to Odoo 14 cache changes

JasminSForgeFlow committed:
[MIG] base_changeset: Migration to 15.0

StefanRijnhart committed:
[FIX] base_changeset: fix caching

ormcache is used to minimize the number of times that the active changeset
rules are quered, but it was still missing in one important place.

In this module's tests, this reduces the number of calls to
`models_to_track_changeset` from 354 to 135 times, and the number of test
queries is reduced from 4069 to 3810.

Commit:
[UPD] Update base_remote.pot

OCA-git-bot committed:
[UPD] README.rst

StefanRijnhart committed:
[FIX] base_changeset: no change for empty values on create

Regression of previous commit

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-15.0/server-tools-15.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-base_remote/

ypapouin committed:
[ADD] database_cleanup: Purge fields (ir.model.fields) added

Commit:
[UPD] Update base_remote.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-15.0/server-tools-15.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-base_remote/

MrGigSolutions committed:
[IMP] base_changeset pre-commmit stuff

MrGigSolutions committed:
[MIG] base_changeset: Migration to 16.0

Co-authored-by stefan@opener.amsterdam
Co-authored-by shams.mukhibillaev@emesa.nl
Co-authored-by remy@emesa.nl

aisopuro committed:
[FIX] sentry: respect sentry_logging_level

Before this fix, the Sentry module sent events for WARNING-
level logs, even if sentry_logging_level was registered as
"error" or higher.

The fix itself is minor: setup of the integration mistakenly
set the hardcoded WARNING level to the event handler and the
sentry_logging_level to the breadcrumb handler, when they
should have been the other way around.

The largest part of the diff is a reworking of the tests in
order to properly replicate the issue:

* The test previously emitted a fake log event directly using
  the integration's handler's emit-method, which skipped the
  part of the logic that actually filters based on logging level.
  This has been changed to use a bespoke NoopHandler and dedicated
  Logger, so that the tests can emit "actual logs" and test Sentry
  as accurately as possible.
* The tests were not configured to use a non-default logging level,
  thus making it so that none of them caught the fact we were basically
  hard-coding the setting to WARNING-level.
  The tests now set the logging level to ERROR in order to make sure
  the configuration parameter works when it is non-default.
* Changes to configuration (especially ignored loggers) were leaking
  from one test into others. The tests were directly mutating the
  `odoo.tools.config.options` mapping, without resetting it afterward,
  leaving the changes in place for subsequent tests.
  Introduced a helper method `patch_config` that can be used to patch
  the config object so that the patch is undone at the end of the test.

aisopuro committed:
[IMP] sentry: increase test coverage

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: server-tools-15.0/server-tools-15.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-15-0/server-tools-15-0-base_remote/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 2.5% (2 of 78 strings)

Translation: server-tools-10.0/server-tools-10.0-fetchmail_attach_from_folder
Translate-URL: https://translation.odoo-community.org/projects/server-tools-10-0/server-tools-10-0-fetchmail_attach_from_folder/it/

NL66278 committed:
[IMP] fetchmail_attach_from_folder: pre-commit stuff

NL66278 committed:
[MIG] fetchmail_attach_from_folder: Migration to 16.0

NL66278 committed:
[MIG] fetchmail_*: adapt tests

NL66278 committed:
[FIX] fetchmail_*: adapt to Odoo 16.0 and python 3.x

vincent-hatakeyama committed:
[IMP] module_change_auto_install: also allow using environment variables

Commit:
[IMP] tracking_manager: allow tracking readonly fields

ypapouin committed:
[FIX] database_cleanup: Check that the model is automatically managed by Odoo

hugosantosred committed:
[FIX] Tests: Use logging.ERROR instead of warn

Sentry level is error by default and warnings are not captured.

rpinset commented:
@rvalyi I followed the wiki though https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#how-to

these are my only modifications
https://github.com/rpinset/server-tools/commit/b5eb35a138a874f142029d6d3879c393644d501a

xtanuiha commented on auto_backup/__manifest__.py:
```diff
@@ -0,0 +1,28 @@
+{
+    "name": "Database Auto-Backup",
+    "summary": "Backups database",
+    "version": "16.0.1.0.0",
```
version should be "17.0.1.0.0"

rpinset replied:
My bad
I forget look this
Thanks 
It's fixed

CRogos reviewed (commented):
Could you solve the pre-commit issues? 
You can also squash translation commits of the same person & language:
![image](https://github.com/OCA/server-tools/assets/1799080/1620099a-3f72-4874-9432-42c7cf556fda)

CRogos committed:
[IMP] remove python libs included in odoo  requirements

florian-dacosta committed:
[FIX] autovacuum_message_attachment : fix inheriting_model feature

OCA-git-bot committed:
Merge PR #2897 into 16.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2813 into 16.0

Signed-off-by Kev-Roche

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2901 into 16.0

Signed-off-by etobella

Commit:
[UPD] Update autovacuum_message_attachment.pot

Commit:
[UPD] Update base_name_search_improved.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-base_name_search_improved
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_name_search_improved/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-autovacuum_message_attachment
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-autovacuum_message_attachment/

OCA-git-bot committed:
Merge PR #2887 into 16.0

Signed-off-by moylop260

OCA-git-bot committed:
[BOT] post-merge updates

astirpe committed:
[MIG] add tests for purge fields

rpinset commented:
> Could you solve the pre-commit issues? You can also squash translation commits of the same person & language: ![image](https://private-user-images.githubusercontent.com/1799080/319107580-1620099a-3f72-4874-9432-42c7cf556fda.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMTY0MjMsIm5iZiI6MTcxMzIxNjEyMywicGF0aCI6Ii8xNzk5MDgwLzMxOTEwNzU4MC0xNjIwMDk5YS0zZjcyLTQ4NzQtOTQzMi00MmM3Y2Y1NTZmZGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTVUMjEyMjAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTc2NGI0M2JjZTA3OGYwNzY0ZDFkNDg4Y2NlMTU5YTQzOWJmMzNhZTA3NzAxYWFmNDAxNmIzYjdjNzRmMWU0YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Up1HITgGBuQmrd1qyDNBoR-3B5HEF9fF-bywkiYLk0E)

Sorry @CRogos, but this is my first attempt to push a migration.
I don't know how to use pre-commit.
I followed the wiki though https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#how-to
these are my only modifications
https://github.com/rpinset/server-tools/commit/b5eb35a138a874f142029d6d3879c393644d501a
and https://github.com/OCA/server-tools/pull/2894/commits/cafb66fb0cbdee05a1969c790ffa459c9cbfd190

CRogos commented:
Yes I also struggled a lot with my first migration. 
pre-commit is a tool to automate certain things like code formatting. in this case, one issue is the xml formatting. 
But instead of changing it manually you should install pre-commit on your dev environment. 
https://github.com/OCA/maintainer-tools/wiki/Install-pre-commit
![image](https://github.com/OCA/server-tools/assets/1799080/c122c6b2-89f2-4416-bbe0-752600a6ef89)

To squash certain commits, you find the instructions here: 
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

in short the most important commands are:
git config --global --add rebase.instructionFormat "(%an <%ae>) %s"
git rebase -i origin/17.0
\# change "pick" by "f" where needed and commit the changes
git push -f

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (88 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/it/

quentinDupont committed:
[16.0][IMP] Update configure for readme

rousseldenis committed:
[IMP] base_exception: Add a hook method / Don't build void exceptions

In order to get exceptions without writing recordset, add a hook method.

Improve exceptions dictionaries building using a defaultdict of a browse
record.

luisDIXMIT committed:
[IMP] base_remote: pre-commit auto fixes

luisDIXMIT committed:
[MIG] base_remote: Migration to 16.0

carolinafernandez-tecnativa committed:
[IMP] datetime_formatter: pre-commit stuff

carolinafernandez-tecnativa committed:
[MIG] datetime_formatter: Migration to 16.0

OCA-git-bot committed:
Merge PR #2913 into 16.0

Signed-off-by dreispt

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2829 into 16.0

Signed-off-by dreispt

Commit:
[UPD] Update bus_alt_connection.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2911 into 16.0

Signed-off-by dreispt

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2918 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update datetime_formatter.pot

OCA-git-bot committed:
[BOT] post-merge updates

florian-dacosta committed:
[IMP] attachment synchronize : when importing files from a remote server :
If one task is failing, keep processing the others
Commit fir attachment imported after each file to be sure not to loose any data, when deleting on remote after import for example

NL66278 committed:
[IMP] fetchmail_.._folder: optionally archive messages automatically

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (52 of 52 strings)

Translation: server-tools-16.0/server-tools-16.0-autovacuum_message_attachment
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-autovacuum_message_attachment/es_AR/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: server-tools-16.0/server-tools-16.0-datetime_formatter
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-datetime_formatter/it/

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
[IMP] update dotfiles

StefanRijnhart committed:
[IMP] base_changeset: hide smart button when there have been no changes

StefanRijnhart committed:
[IMP] base_changeset: refresh screenshots

astirpe committed:
[MIG] adapt code for V16

NL66278 committed:
[IMP] fetch..._folder: Add contributor and maintainer

NL66278 committed:
[FIX] fetchmaeil..folder: adapt test to current demo data

NL66278 committed:
[FIX] fetchmail..folder: handle non standard matching

Match algorithms other then 'odoo_standard' were not correctly dealt with in the migrated code. For
these algorithms we restore the handling (but with improvements) that was present in the 10.0 version.

OCA-git-bot committed:
Merge PR #2923 into 16.0

Signed-off-by bguillot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2800 into 16.0

Signed-off-by gurneyalex

Commit:
[UPD] Update fetchmail_attach_from_folder.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-fetchmail_attach_from_folder
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-fetchmail_attach_from_folder/

renda-dev committed:
[FIX] base_view_inheritance_extension: parse old value

OCA-git-bot committed:
Merge PR #2944 into 16.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

SilvioC2C committed:
[16.0][ADD] server_action_logging

SilvioC2C committed:
[16.0][ADD] base_force_record_noupdate

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: server-tools-16.0/server-tools-16.0-module_auto_update
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-module_auto_update/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 8.0% (2 of 25 strings)

Translation: server-tools-16.0/server-tools-16.0-base_name_search_improved
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_name_search_improved/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 70.0% (7 of 10 strings)

Translation: server-tools-16.0/server-tools-16.0-scheduler_error_mailer
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-scheduler_error_mailer/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 79.6% (43 of 54 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (10 of 10 strings)

Translation: server-tools-16.0/server-tools-16.0-scheduler_error_mailer
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-scheduler_error_mailer/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (76 of 76 strings)

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/pt_BR/

OCA-git-bot committed:
Merge PR #2946 into 16.0

Signed-off-by gurneyalex

Commit:
[UPD] Update base_force_record_noupdate.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2936 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update database_cleanup.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/

OCA-git-bot committed:
Merge PR #2951 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update database_cleanup.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (36 of 36 strings)

Translation: server-tools-16.0/server-tools-16.0-module_analysis
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-module_analysis/it/

OCA-git-bot committed:
Merge PR #2663 into 16.0

Signed-off-by hbrunn

Commit:
[UPD] Update attachment_synchronize.pot

Commit:
[UPD] Update attachment_queue.pot

Commit:
[UPD] Update auto_backup.pot

Commit:
[UPD] Update autovacuum_message_attachment.pot

Commit:
[UPD] Update base_changeset.pot

Commit:
[UPD] Update base_exception.pot

Commit:
[UPD] Update excel_import_export.pot

Commit:
[UPD] Update excel_import_export_unidecode.pot

Commit:
[UPD] Update fetchmail_attach_from_folder.pot

Commit:
[UPD] Update excel_import_export_demo.pot

Commit:
[UPD] Update iap_alternative_provider.pot

Commit:
[UPD] Update rpc_helper.pot

Commit:
[UPD] Update scheduler_error_mailer.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-iap_alternative_provider
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-iap_alternative_provider/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export_demo/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-scheduler_error_mailer
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-scheduler_error_mailer/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-attachment_synchronize
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-attachment_synchronize/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export_unidecode
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export_unidecode/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-rpc_helper
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-rpc_helper/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-auto_backup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auto_backup/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-attachment_queue
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-attachment_queue/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-autovacuum_message_attachment
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-autovacuum_message_attachment/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-fetchmail_attach_from_folder
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-fetchmail_attach_from_folder/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: server-tools-16.0/server-tools-16.0-base_force_record_noupdate
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_force_record_noupdate/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (63 of 63 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (87 of 87 strings)

Translation: server-tools-16.0/server-tools-16.0-auto_backup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auto_backup/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (76 of 76 strings)

Translation: server-tools-16.0/server-tools-16.0-attachment_synchronize
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-attachment_synchronize/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (85 of 85 strings)

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (112 of 112 strings)

Translation: server-tools-16.0/server-tools-16.0-attachment_queue
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-attachment_queue/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (203 of 203 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: server-tools-16.0/server-tools-16.0-autovacuum_message_attachment
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-autovacuum_message_attachment/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (45 of 45 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export_demo/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (19 of 19 strings)

Translation: server-tools-16.0/server-tools-16.0-scheduler_error_mailer
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-scheduler_error_mailer/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: server-tools-16.0/server-tools-16.0-rpc_helper
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-rpc_helper/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: server-tools-16.0/server-tools-16.0-iap_alternative_provider
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-iap_alternative_provider/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (15 of 15 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export_unidecode
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export_unidecode/it/

OCA-git-bot committed:
Merge PR #2811 into 16.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (85 of 85 strings)

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 20.6% (18 of 87 strings)

Translation: server-tools-16.0/server-tools-16.0-auto_backup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auto_backup/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (19 of 19 strings)

Translation: server-tools-16.0/server-tools-16.0-scheduler_error_mailer
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-scheduler_error_mailer/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 50.0% (48 of 96 strings)

Translation: server-tools-16.0/server-tools-16.0-fetchmail_attach_from_folder
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-fetchmail_attach_from_folder/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 77.7% (49 of 63 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (63 of 63 strings)

Translation: server-tools-16.0/server-tools-16.0-base_exception
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_exception/pt_BR/

OCA-git-bot committed:
Merge PR #2945 into 16.0

Signed-off-by gurneyalex

Commit:
[UPD] Update server_action_logging.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (88 of 88 strings)

Translation: server-tools-16.0/server-tools-16.0-auditlog
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-auditlog/pt_BR/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: server-tools-16.0/server-tools-16.0-server_action_logging
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-server_action_logging/it/

OCA-git-bot committed:
Merge PR #2915 into 16.0

Signed-off-by etobella

Commit:
[UPD] Update autovacuum_message_attachment.pot

Commit:
[UPD] Update base_remote.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-autovacuum_message_attachment
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-autovacuum_message_attachment/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-base_remote
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-base_remote/

mymage committed:
Added translation using Weblate (Italian)

rpinset commented:
@CRogos I started again and ended up getting all the checks green 😊
can you check this ?

CRogos commented:
yes indeed. 

But could you still squash some commits. e.g. when there are translations of the same person and same language.
![image](https://github.com/OCA/server-tools/assets/1799080/c36d59a9-dfa4-465b-8148-3a66ceb49630)

All migrations commits should be merged and instead of merging 17.0 into this branch you should have rebased this branch. 
![image](https://github.com/OCA/server-tools/assets/1799080/457966c7-a276-42e5-9f09-2777144854f4)

Maybe you can rebase it again to remove the merge commit?

rpinset commented:
My bad, I forgot. I'm going to try that

CRogos commented:
No need to close it... just rebase and update the branch with push force.

rpinset commented:
> No need to close it... just rebase and update the branch with push force.

this was an automatic consequence of my git cli :(
i try again https://github.com/OCA/server-tools/pull/2975

rpinset commented:
this was an automatic consequence of my git cli :(
i try again here https://github.com/OCA/server-tools/pull/2975
